core/services/feeds/service.go

```diff
@@ -524,6 +641,10 @@ func (s *service) RejectSpec(ctx context.Context, id int64) error {
 		return errors.Wrap(err, "orm: job proposal")
 	}
 
+	if proposal.Status == JobProposalStatusDeleted {
```

Comment:
I don't think you need this. I don't see ay harm in allowing them to reject a spec for deleted job proposal